File: src/main/java/net/ddns/rarnold/randomworkoutroutine/config/WebConfiguration.java
package net.ddns.rarnold.randomworkoutroutine.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfiguration implements WebMvcConfigurer {
    @Override
    public void addViewControllers(ViewControllerRegistry registry) {
        registry.addViewController("/{x:[\\w\\-]+}")
                .setViewName("forward:/");
        registry.addViewController("/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}")
                .setViewName("forward:/");
    }
}

File: web/src/main/kotlin/uk/co/rafearnold/randomworkoutroutine/web/config/WebConfiguration.kt
package uk.co.rafearnold.randomworkoutroutine.web.config

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry

@Configuration
open class WebConfiguration : WebMvcConfigurer {
    override fun addViewControllers(registry: ViewControllerRegistry) {
        registry.addViewController("/{x:[\\w\\-]+}")
                .setViewName("forward:/")
        registry.addViewController("/{x:[\\w\\-]+}/")
                .setViewName("forward:/")
        registry.addViewController("/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}")
                .setViewName("forward:/")
    }
}
